### PR TITLE
#1456 Lightweight messages support

### PR DESCRIPTION
**Pull request by zuker**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,10 +39,10 @@
     "gulp-istanbul": "1.1.1",
     "gulp-jasmine": "0.2.0",
     "jasmine-node": "3.0.0",
-    "mocha": "7.1.2",
-    "nock": "10.0.6",
+    "mocha": "^7.1.2",
```
Comment (anton-kotenko) [inline package.json:42]: plz do not use ^ versions

Comment (zuker) [reply package.json:42]: why? it's dev section and we lock file

File: lib/sailor.js [hunk 2]
```diff
@@ -322,29 +346,118 @@ class Sailor {
...
                                 [stepId]: Object.assign({}, _.omit(payload, 'passthrough'))
                             });
                         }
+                    }
+                }
+
+                data.headers = data.headers || {};
+                const { body, passthrough } = data;
+
+                if (settings.EMIT_LIGHTWEIGHT_MESSAGE) {
```
Comment (anton-kotenko) [inline lib/sailor.js:370]: Is it possible to cut new code into smaller functions?
It's quite complicated already want to simplify

Comment (zuker) [reply lib/sailor.js:370]: i've already done what i can with moving logic out of processMessage into runExec

Comment (zuker) [reply lib/sailor.js:370]: futher steps require heavy refactoring of sailor

File: lib/sailor.js [hunk 3]
```diff
@@ -322,29 +346,118 @@ class Sailor {
...
+                    logger.info('Outgoing lightweight is enabled, going to check size.');
+                    const bodyBuf = Buffer.from(JSON.stringify(body), 'utf-8');
+                    const passthroughBufs = Object.keys(passthrough).map(stepId => ({
```
Comment (anton-kotenko) [inline lib/sailor.js:373]: Correct me if I'm wrong, but passthrough might already be in maester. 
So sometimes there are no need to do all this. for passthrough

Comment (zuker) [reply lib/sailor.js:373]: yep, and actual upload does not happen if incoming passthrough has and id header

Comment (zuker) [reply lib/sailor.js:373]: here https://github.com/elasticio/sailor-nodejs/pull/150/files#diff-49130a3beafac89ab92d914a0be244a5R393

**Pull request merged**